### PR TITLE
ECO-645 Update keygen to match Rust Client

### PR DESCRIPTION
**Pull request by sacherjj**
Updating client-py `keygen` and `validator-keygen` to use secret_key.pem, public_key.pem, and public_key_hex naming.
Added prefix byte (01 - ed25519, 02 - secp256k1) to hex for export.

https://casperlabs.atlassian.net/browse/ECO-645

### CODE REVIEW

**Review by momipsl — APPROVED** [review]
This all seems fine to me - I have no issue with the implementation.  

However I do question the fact that from a single private key, the simplified mechanics of account derivation mean that we can only project to a single account, furthermore the projected account is non-chain specific.  

I.E. imagine a situation whereby an entity has generated a key, they then use that key on test net.  Sometime after they can use it upon main net.  This causes ambiguity and potential confusion.  Most wallets will generate a seed, 32 random bytes, and then derive keys for target nets.  The derived keys are isolated from each other and thus the ambiguity is removed.  The derivation path normally includes at least the chain identifier.  

So in summary:  the implementation is a good one & thus approved, however there are some wider questions regarding keys that need to be borne in mind.
